client/flutter/lib/pages/onboarding/notifications_page.dart

```diff
@@ -25,6 +31,19 @@ class _NotificationsPageState extends State<NotificationsPage> {
   }
 
   void _allowNotifications() async {
+    // iOS needs a call for permissions but Android doesn't.
+    // If the user opts-out on Android we just don't register the device token.
+    await _firebaseMessaging.requestNotificationPermissions(
```

Comment:
this part of the `firebase_messaging` package is pretty poorly documented, and [their example](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_messaging/example/lib/main.dart) doesn't use `async`/`await` at all which seems wrong... but wanted to confirm that you've tested the different states of what can happen here and we don't need to handle any errors. E.g. if a user rejects the system dialog, it won't throw an error that isn't caught here, and/or the `getToken()` call doesn't depend on the user accepting the permission request